### PR TITLE
PLT-7407: Add client4 apis needed for JIRA plugin

### PR DESCRIPTION
**Pull request by ccbrown**
#### Summary
* ~Back-end plugin mechanism~ https://github.com/mattermost/platform/pull/7177
* ~Implement Windows IPC~ https://github.com/mattermost/platform/pull/7251
* ~Make hook methods optional (ideally without a roundtrip for unimplemented methods)~ https://github.com/mattermost/platform/pull/7263
* **Add APIs required for JIRA plugin (HTTP route registration and post creation)** (along with https://github.com/mattermost/platform/pull/7289)
* Port JIRA plugin (this'll be almost entirely build system changes once the above are done)

Very straight-forward and easy to add these APIs. `net/rpc` does all the heavy lifting. But it is a little tedious. It's certainly feasible to eliminate the tedium with some code generation.

Once @jwilander's PR (https://github.com/mattermost/platform/pull/7279) is merged, the final piece will be implementing the "API provider" to tie the backend plugins and `app` together. Then the fun begins! (Actually developing the backend plugins.)

#### Ticket Link
Part of https://mattermost.atlassian.net/browse/PLT-7407

#### Checklist
- [x] Added or updated unit tests (required for all new features)
- [x] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/platform/blob/master/i18n/en.json) and [.../webapp/i18n/en.json](https://github.com/mattermost/platform/tree/master/webapp/i18n/en.json)) updates

### CODE REVIEW

File: plugin/plugintest/api.go [hunk 1]
```diff
@@ -15,3 +16,43 @@ var _ plugin.API = (*API)(nil)
 func (m *API) LoadPluginConfiguration(dest interface{}) error {
 	return m.Called(dest).Error(0)
 }
+
+func (m *API) GetTeamByName(name string) (*model.Team, *model.AppError) {
+	ret := m.Called(name)
```
Comment (enahum) [inline plugin/plugintest/api.go]: Can we name `m` as `api` ? 0/5 on this is

File: plugin/rpcplugin/api.go [hunk 2]
```diff
@@ -26,6 +28,67 @@ func (h *LocalAPI) LoadPluginConfiguration(args struct{}, reply *[]byte) error {
...
+	Team  *model.Team
+	Error *model.AppError
+}
+
+func (api *LocalAPI) GetTeamByName(args string, reply *APITeamReply) error {
+	team, err := api.api.GetTeamByName(args)
+	*reply = APITeamReply{
+		Team:  team,
+		Error: err,
+	}
```
Comment (jwilander) [inline plugin/rpcplugin/api.go]: Why the structs and the setting of pointers instead of just returning the team and error?

Comment (ccbrown) [reply plugin/rpcplugin/api.go]: This is just how `net/rpc` works. RPC methods have to look like this:

```go
func (t *T) MethodName(argType T1, replyType *T2) error
```

See https://golang.org/pkg/net/rpc/

**Comment by jwilander** [comment]
Needs a rebase, but otherwise ready to merge

**Pull request merged**